x/k6/baggage.go

```diff
@@ -31,15 +55,15 @@ func (lh *labelHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	labels := getBaggageLabels(r)
+	ctx := r.Context()
+	labels := getBaggageLabelsFromContext(ctx)
 	if labels == nil {
 		lh.innerHandler.ServeHTTP(w, r)
 		return
 	}
```

Comment:
I might be mistaken (couldn't find a test case), but it seems that `baggageToLabels` called from `getBaggageLabelsFromContext` may return a non-nil empty `labels` (`LabelSet`), if the baggage only contains non-k6 labels. I would move zero length check to `baggageToLabels`, so that we do this after filtering, just in case

Reply:
Hmm I'm not sure I completely follow.

You are right that if a baggage header came in but all members were filtered out, we'd end up with a non-nil, empty `LabelSet` at this point. In which case, we'd fall through and try set labels in the profile context. I fixed that issue in https://github.com/grafana/pyroscope-go/pull/133/commits/8477ab7c8f986fd8556ba561326d1616889b2a86. But I'm not sure I addressed your original concern.